Fix linting issues in utils

For all of [these issues](https://github.com/feder-observatory/stellarphot/issues?q=is%3Aopen+is%3Aissue+label%3Alinting), first delete the appropriate line in `pyproject.toml` that excludes the subpackage from checks. For `utils` you would delete [this line](https://github.com/feder-observatory/stellarphot/blob/main/pyproject.toml#L127):

```toml
"stellarphot/utils/*" = ["F", "ARG"]
```

After that, run `ruff` either with `tox -e lint` or `ruff .`. The errors you get should be the ones -- fix them! If it is an unused function argument make sure to search the project for calls to that function to update the arguments there too.

- [ ] stellarphot/utils/magnitude_transforms.py:347:9: F841 Local variable `dy` is assigned to but never used
- [ ] stellarphot/utils/magnitude_transforms.py:517:5: ARG001 Unused function argument: `obs_filter`
- [ ] stellarphot/utils/magnitude_transforms.py:529:5: ARG001 Unused function argument: `verbose`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -121,9 +121,6 @@ select = [
 # Ignore F405 (variable may be from star imports) in docs/conf.py
 "docs/conf.py" = ["F405"]
 
-# Ignore pyflakes and ARGS issues in each subpackage until they are fixed
-"stellarphot/utils/*" = ["F", "ARG"]
-
 [tool.pytest.ini_options]
 minversion = 7.0
 testpaths = [

--- a/stellarphot/utils/comparison_utils.py
+++ b/stellarphot/utils/comparison_utils.py
@@ -7,7 +7,6 @@ from astropy.coordinates import SkyCoord
 from astropy.nddata import CCDData
 from astropy import units as u
 
-from stellarphot.photometry import *
 from stellarphot import apass_dr9, vsx_vizier
 
 

--- a/stellarphot/utils/magnitude_transforms.py
+++ b/stellarphot/utils/magnitude_transforms.py
@@ -304,13 +304,13 @@ def calculate_transform_coefficients(
     mag_diff = catalog_mag - (input_mag - 2.5 * np.log10(gain))
 
     # The error is the errors of those combined in quadrature.
-    combined_error = np.sqrt(input_mag_error**2 + catalog_mag_error**2)
+    # combined_error = np.sqrt(input_mag_error**2 + catalog_mag_error**2)
 
     # If both errors are zero then the error is omitted.
-    if (combined_error == 0).all():
-        dy = None
-    else:
-        dy = combined_error
+    # if (combined_error == 0).all():
+    #     dy = None
+    # else:
+    #     dy = combined_error
 
     g_init = models.Polynomial1D(order)
     fit = fitting.LinearLSQFitter()
@@ -558,7 +558,7 @@ def transform_to_catalog(
 
     """
 
-    if obs_error_column is None:
+    if obs_error_column is None and verbose:
         print("are you sure you want to do that? Error weighting is important!")
 
     fit_bounds_lower = [
@@ -595,9 +595,10 @@ def transform_to_catalog(
     cat = None
     cat_coords = None
 
-    for file, one_image in zip(
+    for file, one_image_inp in zip(
         observed_mags_grouped.groups.keys, observed_mags_grouped.groups
     ):
+        one_image = one_image_inp[one_image_inp["passband"] == obs_filter]
         our_coords = SkyCoord(one_image["RA"], one_image["Dec"], unit="degree")
         if cat is None or cat_coords is None:
             cat = apass_dr9(

